### DISCUSSION TITLE
Veins build error

### DISCUSSION THREAD

**Discussion by Aravindhan-Mathivanan** [omnetpp/omnetpp#1235] [General]
Hi, 
when i build my veins on omnetpp i'm getting this error

>> referenced by veins/modules/mobility/traci/TraCIScenarioManagerForker.cc:150
>>>               ../out/gcc-debug/src/veins/modules/mobility/traci/TraCIScenarioManagerForker.o:(veins::TraCIScenarioManagerForker::getPortNumber() const)
collect2.exe: error: ld returned 1 exit status
C:/Users/Aravindhan/Downloads/omnetpp-6.0.3/tools/win32.x86_64/mingw64/bin/llvm-ar.exe: error: unable to load '../out/gcc-debug/src/libveins_dbg.dll.a': No such file or directory
make[1]: *** [Makefile:217: ../out/gcc-debug/src/libveins_dbg.dll] Error 1
make[1]: Leaving directory '/c/Users/Aravindhan/Downloads/veins-veins-5.2/src'
make: *** [Makefile:36: all] Error 2
"make MODE=debug -j12 all" terminated with exit code 2. Build might be incomplete.

**Comment by avarga** [comment]
I am not sure what is the problem in your case, but on Linux (or on Windows, in WSL) you can install Veins with SUMO along with a matching version of INET and OMNeT++ *with a single command*:

    opp_env install -w veins-workspace --init veins-latest

Then you can go grab some coffee or tea and enjoy it while `opp_env` downloads the packages, unpacks them, edits config files so projects find each other, and compiles the projects. Then change into the new `veins-workspace/`  directory, and enter

    opp_env shell

to get a shell prompt where you can work with Veins.

You can grab `opp_env` from here: https://github.com/omnetpp/opp_env#opp_env-automated-installation-of-omnet-simulation-frameworks.
